backend/cmm_helpers.ml

```diff
@@ -2299,9 +2299,12 @@ let apply_function_body (arity, (mode : Lambda.alloc_mode)) =
   (* In the slowpath, a region is necessary in case the initial applications do
      local allocations *)
   let region =
-    match mode with
-    | Alloc_heap -> Some (V.create_local "region")
-    | Alloc_local -> None
+    if not Config.stack_allocation
+    then None
+    else
+      match mode with
+      | Alloc_heap -> Some (V.create_local "region")
+      | Alloc_local -> None
```

Comment:
This can't be the fix: it's only possible to construct the value `Alloc_local` if Config.stack_allocation is true. (see lambda.ml) Are you seeing `Alloc_local` somewhere when stack_allocation is false?